packages/itwinui-react/src/core/Tooltip/Tooltip.test.tsx

```diff
@@ -39,14 +39,22 @@ it('should be visible', () => {
   getByText('some text');
 });
 
-it('should not be visible', () => {
-  const { queryByText } = render(
+it('should respect visible prop', () => {
+  const { queryByText, rerender } = render(
     <Tooltip content='some text' visible={false}>
       <div>Visible!</div>
     </Tooltip>,
   );
 
-  expect(queryByText('some text')).toBeNull();
+  expect(queryByText('some text')).not.toBeVisible();
+
+  rerender(
+    <Tooltip content='some text' visible>
+      <div>Not visible!</div>
+    </Tooltip>,
+  );
+
+  expect(queryByText('some text')).toBeVisible();
```

Comment:
I guess the `<div>` contents on L45/54 ("Visible" / "Not visible") are swapped

```suggestion
it('should respect visible prop', () => {
  const { queryByText, rerender } = render(
    <Tooltip content='some text' visible={false}>
      <div>Not visible!</div>
    </Tooltip>,
  );

  expect(queryByText('some text')).not.toBeVisible();

  rerender(
    <Tooltip content='some text' visible>
      <div>Visible!</div>
    </Tooltip>,
  );

  expect(queryByText('some text')).toBeVisible();
```